staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -1512,8 +1512,46 @@ func RunWatchSemantics(ctx context.Context, t *testing.T, store storage.Interfac
 				require.NoError(t, err, "failed to add a pod: %v")
 				createdPods = append(createdPods, out)
 			}
-			testCheckResultsInStrictOrder(t, w, scenario.expectedEventsAfterEstablishingWatch(createdPods))
-			testCheckNoMoreResults(t, w)
+
+			// before this pr:
+			// Bookmark event is only generated by Cacher.dispatchEvents goroutine. And it will re-calculate the next time when bookmark should be sent.
+			// DispatchEvents goroutine will set bookmark event nextTime from immediately to defaultBookmarkFrequency(1min) after sending initial events end bookmark event.
+			// So reflector in client will not receive another bookmark event within 1 minute after receiving the initial events end bookmark event.
+			// And client can receive only non-bookmark event within 1 minute.
+
+			// after this pr:
+			// Initial events end bookmark event is generated by cacheWatcher.processInterval goroutine. And ordinary bookmark event is still generated by Cacher.dispatchEvents goroutine.
+			// After sending all events in cacherInterval, cacheWatcher.processInterval goroutine tries to send initial events end bookmark event. Before cacheWatcher.state is set to cacheWatcherBookmarkSent,
+			// Cacher.dispatchEvents goroutine may re-calculate the next time when bookmark should be sent. And it will still believe that bookmark should be sent immediately.
+			// Within the 1s-1.25s bookmark tick time, current unit test creates pods defined in scenario.podsAfterEstablishingWatch field.
+			// When bookmark tick is triggerred in 1s-1.25s , apiserver may send a non-initial bookmark event(resourceVersion can be equal to any pods' resourceVersion defined in scenario.podsAfterEstablishingWatch) to
+			// client. So we can not make sure that client will not receive another bookmark event within 1 minute after receiving the initial events end bookmark event.
+
+			// In a word, this phenomenon doesn't affect the function of watchlist feature. It's just that we need to pay attention to this phenomenon when writing ut cases
+
+			// get corresponding pod watch events
```

Comment:
I initially started thinking if making bookmark period a configurable option and switching it off wouldn't be a better option for this test, but in the end, I wouldn't go this way.

That said, I would approach this slightly differently. In particular:
- let's introduce `testCheckResultWithIgnoreFunc` function
  it's supposed to be somewhat similar to `testCheckResultFunc`, but additionally takes `ignore func(watch.Event) bool` parameter [and it has a loop for getting items from watch and just ignores one for which the `ignore` function returns true)
- let's simply use this new helper function here, passing function that ignores events with type Bookmark as the `ignore` function